torch/csrc/jit/passes/symbolic_shape_analysis.cpp

```diff
@@ -529,13 +533,235 @@ struct SymbolicShapeNodeAnalyzer {
   Node* node_;
 };
 
-void PropagateShapesWithShapeFunction(
+std::shared_ptr<Graph> PropagateShapesWithShapeFunction(
     Node* n,
     std::shared_ptr<Graph>& shape_compute_graph,
     const AliasDb& db) {
-  SymbolicShapeNodeAnalyzer(n, shape_compute_graph, db).run();
+  return SymbolicShapeNodeAnalyzer(n, shape_compute_graph, db).run();
 }
 
+struct SymbolicShapeGraphAnalyzer {
+  SymbolicShapeGraphAnalyzer(
+      std::shared_ptr<Graph>& graph,
+      Node* beg,
+      Node* end)
+      : graph_(graph), beg_(beg), end_(end) {
+    TORCH_INTERNAL_ASSERT(
+        beg_->owningBlock() == end_->owningBlock() && end_->isAfter(beg_));
+  }
+
+  c10::optional<ShapeComputeGraphMapping> run() {
+    AliasDb db(graph_);
+    std::unordered_map<Node*, std::shared_ptr<Graph>> partial_evaluated_graphs =
+        propagateShapesAndGatherPartialEvalShapeGraphs(db);
+
+    auto large_shape_compute_graph = std::make_shared<Graph>();
```

Comment:
General q: does "large" have a specific meaning in this diff?  It seems like it refers to the shape graph formed by joining multiple nodes' shape graphs, is that right?

Reply:
yeaaaa.... naming suggestions would be appreciated 😃  maybe Stitched ? Joined ?

Reply:
Haha, yeah, stitched and joined are both good and descriptive :-)